### PR TITLE
feat: introduce react-native-sn-table

### PR DESCRIPTION
**Pull request by vcellu**
### This PR introduces the react-native-sn-table.
It uses all of the logic from the supernova, but renders the table in a native fashion.

The plan for releasing is to release two separate packages.  One for sn-table and one for react-native-sn-table. Nebula.js build has been updated to accept a flag "--reactNative".  This will signal to rollup to pick up all .native files in favour of the plain files.  For example, if there are two files a.jsx and a.native.jsx, passing the "--reactNative" flag will pick up the a.native.jsx.  Not passing the flag will ignore a.native.jsx

### CODE REVIEW

**Review by niekvanstaveren — COMMENTED** [review]
I guess the setup works. However it does not look like there is any code shared between the desktop and native variant. If that stays the case what is the need of having this in the same repo?

File: package.json
Comment (haxxmaxx) [reply package.json:78]: We've fixed this, this package.json is outdated

File: package.json
Comment (vcellu) [reply package.json]: Changed to reactNative

File: react-native/package.json
Comment (vcellu) [reply react-native/package.json]: Done

**Comment by niekvanstaveren** [comment]
@Caele Could you have a look at this as well?

**Review by Caele — APPROVED** [review]
I'm a bit rusty, but don't see anything strange here, looks good.

**Review by niekvanstaveren — APPROVED** [review]

**Comment by haxxmaxx** [comment]
The tests have passed but are failing to upload the coverge report. This is due to being on a fork, we haven't fixed this yet. I can force merge this if you want

File: src/handle-data.js
Comment (LiKang6688) [inline src/handle-data.js:48]: why `pageInfo` is needed here?

File: src/handle-data.js
Comment (LiKang6688) [inline src/handle-data.js:53]: same concern here

**Comment by LiKang6688** [comment]
Hi @vcellu, I have the same question as Niek. I do not see that react native table shares any code with the sn-table. Why do we want to put them together? Please correct me if I was wrong.

**Comment by LiKang6688** [comment]
> Hi @vcellu, I have the same question as Niek. I only see that react native table shares `sortingFactory` and `manageData`, these two functions with the sn-table. Why do we want to put them together? Please correct me if I was wrong.

**Comment by LiKang6688** [comment]
Hi @vcellu , I think it would be very nice if you could add some docs, such as how do we run, build, test, deploy, and develop the react-native-sn-table. Something like https://github.com/qlik-oss/sn-table/blob/main/.github/CONTRIBUTION.md and https://github.com/qlik-oss/sn-table/blob/main/README.md.

File: src/handle-data.js
Comment (vcellu) [reply src/handle-data.js:53]: I didn't add the code for the MAX_CELLS, I just added the code for && pageInfo.

File: src/handle-data.js
Comment (vcellu) [reply src/handle-data.js:48]: This is an infinite scroll signal, that there is no data.  So pagInfo is set to null to avoid the re-renders.

Comment (LiKang6688) [reply src/handle-data.js:48]: I see. Thanks.

**Review by LiKang6688 — APPROVED** [review]

**Comment by vcellu** [comment]
> 

We are waiting for a build PR to get merged.  Once that is complete we will create a full document.   On how to build and deploy.  Currently you just run `"build:rn": "yarn run locale:generate && node ./tools/build.js --reactNative".  But the full documentation will be released when we have merged the build tools into nebula.js

**Pull request merged**
